addons/pkg/util/tkr_util.go

```diff
@@ -38,7 +44,7 @@ func GetTKRByNameV1Alpha1(ctx context.Context, c client.Client, tkrName string)
 }
 
 // GetTKRByNameV1Alpha3 gets v1Alpha3 TKR object given a TKR name
-func GetTKRByNameV1Alpha3(ctx context.Context, c client.Client, tkrName string) (*runtanzuv1alpha3.TanzuKubernetesRelease, error) {
+func GetTKRByNameV1Alpha3(ctx context.Context, c client.Client, cluster *clusterapiv1beta1.Cluster, tkrName string) (*runtanzuv1alpha3.TanzuKubernetesRelease, error) {
 	tkrV1Alpha3 := &runtanzuv1alpha3.TanzuKubernetesRelease{}
```

Comment:
I am still not comfortable with changing a published function. But I guess no one else is using GetTKRByNameV1Alpha3 but us, so we could get away with it.